src/sisl/io/wannier90/seedname.py

```diff
@@ -211,21 +199,13 @@ def read_geometry(self, *args, **kwargs):
         """Reads a `Geometry` and creates the Wannier90 cell"""
 
         # Read in the super-cell
-        lattice = self.read_lattice()
-
-        self._set_file("_centres.xyz")
-        if self.file.is_file():
-            geom = self._r_geometry_centres()
-        else:
-            self._set_file()
-            geom = self._r_geometry(lattice, *args, **kwargs)
-
-        # Reset file
-        self._set_file()
+        kwargs["lattice"] = self.read_lattice()
```

Comment:
this won't allow users to put in their own lattice (for whatever reason).
Test for `lattice` in `kwargs` if not, read it, otherwise retain.

Reply:
should be fixed